awscli/customizations/commands.py

```diff
@@ -134,7 +134,7 @@ def __call__(self, args, parsed_globals):
         event = 'before-building-argument-table-parser.%s' % \
             ".".join(self.lineage_names)
         self._session.emit(event, argument_table=self._arg_table, args=args,
-                           session=self._session)
+                           session=self._session, parsed_globals=parsed_globals)
```

Comment:
since `update_endpoint_url` runs every time this event is emitted, `parsed_globals` needs to be passed here as well for the command not to fail/return a 255